spinoso-time/src/time/tzrs/to_a.rs

```diff
@@ -0,0 +1,78 @@
+use super::Time;
+
+/// Serialized representation of a timestamp using a ten-element array of
+/// datetime components.
+///
+/// [sec, min, hour, day, month, year, wday, yday, isdst, zone]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ToA {
+    /// The second of the minute `0..=59` for the source _time_.
+    pub sec: u8,
+    /// The minute of the hour `0..=59` for the source _time_.
+    pub min: u8,
+    /// The hour of the day `0..=23` for the source _time_.
+    pub hour: u8,
+    /// The day of the month `1..=n` for the source _time_.
+    pub day: u8,
+    /// The month of the year `1..=12` for the source _time_.
+    pub month: u8,
+    /// The year (including the century) for the source _time_.
+    pub year: i32,
+    /// An integer representing the day of the week, `0..=6`, with Sunday == 0
+    /// for the source _time_.
+    pub wday: u8,
+    /// An integer representing the day of the year, `1..=366` for the source
+    /// _time_.
+    pub yday: u16,
+    /// Whether the source _time_ occurs during Daylight Saving Time in its time
+    /// zone.
+    pub isdst: bool,
+    /// The timezone used for the source _time_.
+    pub zone: String,
```

Comment:
Yes, an owned string, for a couple of reasons:

- `Offset::Fixed` generates a string for the `time_zone_designation` during runtime. I'm not sure if there's some fancy way to make this into a `&'static`. We do know that `time_zone_designation` is a fixed width u8 buffer, https://docs.rs/tz-rs/0.6.9/src/tz/timezone/mod.rs.html#83, so maybe there's some magic there?
- If we can't use `&'static`, then it's possible `Time` is dropped after creating a `ToA`, so it would make sense to own it instead of specifying a dependent lifetime

I'm all ears if there's another way to do this one of course

Reply:
We can use a `Cow<'static, str>` but I'm not sure it's worth the code complexity to save an allocation. What do you think @b-n?

Reply:
That would mean the functions that use this would have to check if it's a `Cow::Borrowed` or a `Cow::Owned` right? If so, I think that's unneeded complexity. Assuming 99% of the use cases are Ruby, then

```
irb(main):001:0> t = Time.now
=> 2022-06-25 20:25:09 +0200
irb(main):002:0> a = t.to_a
=> [9, 25, 20, 25, 6, 2022, 6, 176, true, "CEST"]
irb(main):003:0> a[9] = "HELLO"
=> "HELLO"
irb(main):004:0> t.zone
=> "CEST"
```

^-- e.g. we're note expecting `a[9]` to be it's own alloc'd value and not a ref.

Reply:
> That would mean the functions that use this would have to check if it's a Cow::Borrowed or a Cow::Owned right?

Not quite: `Cow<'static, str>` derefs to `&str` and can be turned into a `String` with `Cow::into_owned`.

Agreed though that this doesn't seem worth it.